subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultProjectLayout.java

```diff
@@ -18,41 +18,31 @@
 
 import org.gradle.api.Project;
 import org.gradle.api.Task;
-import org.gradle.api.Transformer;
 import org.gradle.api.file.ConfigurableFileCollection;
 import org.gradle.api.file.Directory;
 import org.gradle.api.file.DirectoryProperty;
 import org.gradle.api.file.FileCollection;
 import org.gradle.api.file.ProjectLayout;
-import org.gradle.api.file.RegularFile;
 import org.gradle.api.internal.file.collections.MinimalFileSet;
-import org.gradle.api.internal.provider.MappingProvider;
 import org.gradle.api.internal.provider.PropertyHost;
-import org.gradle.api.internal.provider.Providers;
 import org.gradle.api.internal.tasks.TaskDependencyFactory;
-import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.util.PatternSet;
 import org.gradle.internal.Factory;
 
 import java.io.File;
 
-public class DefaultProjectLayout implements ProjectLayout, TaskFileVarFactory {
+public class DefaultProjectLayout extends AbstractFileSystemLayout implements ProjectLayout, TaskFileVarFactory {
```

Comment:
Maybe we could consider to use composition instead of inheritance here? Then definition of `DefaultProjectLayout` can remain old and class hierarchy will be simpler.

Reply:
That would require introducing a new abstraction that we could delegate to, right? I am a fan of composition for better separating responsibilities, but feel in this case not much would be left to `ProjectLayout` itself.

Reply:
Yup, thanks for the example, that was what I was thinking more or less.